lib/core/test/unit/Cardano/Wallet/Primitive/AddressDiscoverySpec.hs

```diff
@@ -312,6 +321,32 @@ prop_lookupDiscovered (s0, addr) =
             expectationFailure "couldn't find private key corresponding to addr"
 
 
+{-------------------------------------------------------------------------------
+                        Properties for CompareDiscovery
+-------------------------------------------------------------------------------}
+
+prop_compareKnownUnknown
+    :: (SeqState DummyTarget, ShowFmt Address, ShowFmt Address)
+    -> Property
+prop_compareKnownUnknown (s, ShowFmt known, ShowFmt addr) =
+    case (fst $ isOurs known s, fst $ isOurs addr s) of
+        (True, False) -> cover 10 True "known-unknown" $ prop LT
```

Comment:
this one along with `prop_compareAntiSymmetric` will check `known-unknow` and `unknow-know`. `know-know` will be covered by the aforementioned property. The missing one is `unknow-unknow` although very obvious one